src/main.rs

```diff
@@ -475,7 +477,11 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         - mem_map.regions().next().unwrap().base().bits();
 
     // Parse the user-mode ELF containing the user-mode task.
-    let umode_bytes = include_bytes!("../target/riscv64gc-unknown-none-elf/release/umode");
+    // Safe, because it comes from the Linker
+    let umode_bin = unsafe { core::ptr::addr_of!(_umode_bin) as *const u8 };
+    let umode_bin_len = unsafe { core::ptr::addr_of!(_umode_bin_len) as usize };
+    let pre_umode_bytes = core::ptr::slice_from_raw_parts::<u8>(umode_bin, umode_bin_len);
+    let umode_bytes = unsafe { pre_umode_bytes.as_ref().unwrap() };
```

Comment:
these two lines can be replaced by a single `core::slice::from_raw_parts`

Reply:
fixed